src/wanderer/guzzlr.ts

```diff
@@ -0,0 +1,101 @@
+import { buy, craftType, Location, print, retrieveItem } from "kolmafia";
+import { $item, get, Guzzlr, have } from "libram";
+import { freeCrafts, globalOptions } from "../lib";
+import { garboValue } from "../session";
+import {
+  canAdventureOrUnlock,
+  DraggableFight,
+  WandererTarget,
+  wandererTurnsAvailableToday,
+} from "./lib";
+
+function considerAbandon(locationSkiplist: Location[]) {
+  const location = Guzzlr.getLocation();
+  const remaningTurns = Math.ceil(
+    (100 - get("guzzlrDeliveryProgress")) / (10 - get("_guzzlrDeliveries"))
+  );
+
+  print(
+    `Got guzzlr quest ${Guzzlr.getTier()} at ${Guzzlr.getLocation()} with remaining turns ${remaningTurns}`
+  );
+
+  if (
+    // consider abandoning
+    !location || // if mafia faled to track the location correctly
+    locationSkiplist.includes(location) ||
+    !canAdventureOrUnlock(location) || // or the zone is marked as "generally cannot adv"
+    (globalOptions.ascending && wandererTurnsAvailableToday(location) < remaningTurns) // or ascending and not enough turns to finish
+  ) {
+    print("Abandoning...");
+    Guzzlr.abandon();
+  }
+}
+
+function acceptGuzzlrQuest(locationSkiplist: Location[]) {
```

Comment:
I think eventually guzzlr will be worth so little that we want to basically only do it to cap guzzlr, which will require some logic changes. Will it be difficult to do that? How do we tell it that doing a bronze has value outside of just the bucks?

I think when we're still charging it up we will want to ignore the cost of booze and of zone unlockers, but when we're done we will want to pay attention to them. That can be out of scope of this, though.